esmvaltool/recipes/recipe_collins13ipcc.yml

```diff
@@ -0,0 +1,3071 @@
+# recipe_collins13ipcc.yml
+
+---
+documentation:
+  description: |
+    Recipe reproducing selected figures from IPCC AR5,
+    chap. 12 (Collins et al., 2013) mainly difference maps between
+    future and present (no observations included), timeseries,
+    changes scaled by mean global temperature change.
+    Only a subset of models are included for most examples!
+    Diagnostic IAV_calc_thetao (zonal ocean temperature, variable thetao) needs
+    a lot of memory, make sure you have enough memory available to read
+    piControl runs or exclude! 
+
+  authors:
+    - lore_ru
+
+  maintainer:
+    - lore_ru
+
+  references:
+    - collins13ipcc
+
+  projects:
+    - crescendo
+
+preprocessors:
+  preproc_map:
+    regrid:
+      target_grid: 1x1
+      scheme: linear
+
+  preproc_map_land:
+    mask_landsea:
+      mask_out: sea
+#    mask_landseaice:
+#      mask_out: ice
+    regrid:
+      target_grid: 1x1
+      scheme: linear
+
+  preproc_atm:
+    regrid:
+      target_grid: 1x1
+      scheme: linear
+    extract_levels:
+      levels: reference_dataset
+      scheme: linear
+
+  preproc_zonal_atm:
+    regrid:
+      target_grid: 1x1
+      scheme: linear
+    extract_levels:
+      levels: reference_dataset
+      scheme: linear
+    zonal_means:
+      coordinate: longitude
+      mean_type: mean
+
+  preproc_ocean:
+    regrid:
+      target_grid: 1x1
+      scheme: linear
+    extract_levels:
+      levels: [5., 15., 25., 35., 45., 56., 66., 77., 88., 100, 112., 125.,
+               140., 155., 170., 190., 215., 245., 275., 315., 370., 430.,
+               510., 610., 725., 870., 1040., 1240., 1450., 1725., 2010.,
+               2315., 2640., 2985., 3340., 3705., 4075., 4460., 4845., 5200.]
+      scheme: linear_horizontal_extrapolate_vertical
+
+  preproc_zonal_ocean:
+    regrid:
+      target_grid: 1x1
+      scheme: linear
+    extract_levels:
+      levels: [5., 15., 25., 35., 45., 56., 66., 77., 88., 100, 112., 125.,
+               140., 155., 170., 190., 215., 245., 275., 315., 370., 430.,
+               510., 610., 725., 870., 1040., 1240., 1450., 1725., 2010.,
+               2315., 2640., 2985., 3340., 3705., 4075., 4460., 4845., 5200.]
+      scheme: linear_horizontal_extrapolate_vertical
+    zonal_means:
+      coordinate: longitude
+      mean_type: mean
+
+diagnostics:
+
+  ###  Mean variable change in one RCP for individual models ##################
+  tas_change_all_models:
+    description: Air temperature change for RCP45 for all models as maps.
+                 Needs historical and scenario run for each model to be plotted.
+    themes:
+      - phys
+    realms:
+      - atmos
+    variables:
+      tas:
+        mip: Amon
+        field: T2Ms
```

Comment:
```suggestion
```

Reply:
Some of my routines need to check if the variable is 2D or 3D, therefore still needed. If thats an issue, the code needs to be changed.

Reply:
You can also do it with:
```
ndim = dimsizes(dimsizes(var))
if (ndim.eq.3) then
...
elseif (ndim.eq.2) then
...
end if
```

This would be a bit more robust, since it checks the actual dimensionality of the variable.

Reply:
I updated all diag_scripts and the recipe and tested it, seems to be ok.

Reply:
Still we need [#170](https://github.com/ESMValGroup/ESMValCore/pull/170) to finalize this, correct?

Reply:
Ideally yes. For the time being there are two lines which are commented out in the recipe which would need to be uncommented once #170 is merged and fixed. AR6 authors are waiting for this, so depending on the time frame of #170 I would prefer not to delay this too much.